### ISSUE TITLE
OneDrive upload fails

### ISSUE THREAD

**Issue by lavinir** [lavinir/hassio-onedrive-backup#105]
### Discussed in https://github.com/lavinir/hassio-onedrive-backup/discussions/104

<div type='discussions-op-text'>

<sup>Originally posted by **Operations8** August 19, 2023</sup>
I am trying to use this nice addon but my upload fails everytime.

Logs shows this:
```
/19/2023 17:47:56 [6] Info: Backup interval configured to every 24 hours
08/19/2023 17:47:56 [6] Info: File Sync Disabled
08/19/2023 17:47:56 [6] Warning: Token cache is empty
08/19/2023 17:47:59 [10] Info: To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code C5ELPSE8X to authenticate.
08/19/2023 17:48:52 [15] Info: Uploading hass_onedrive_backup_2023-08-19-17-41 (08/19/2023 17:41:23)
08/19/2023 17:48:52 [15] Info: Fetching Local Backup (Slug:111d9886)
08/19/2023 17:48:54 [14] Info: Backup (111d9886) fetched successfully
08/19/2023 17:48:55 [10] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: invalidRequest
Message: Invalid request
Inner error:
	AdditionalData:
	date: 2023-08-19T15:48:55
		request-id: 38585621-36e2-439d-b617-76d1e626e888
	client-request-id: 38585621-36e2-439d-b617-76d1e626e888
ClientRequestId: 38585621-36e2-439d-b617-76d1e626e888
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at hassio_onedrive_backup.Graph.GraphHelper.UploadFileAsync(String filePath, DateTime date, String instanceName, TransferSpeedHelper transferSpeedHelper, String destinationFileName, Action`2 progressCallback, Boolean flatten, String description) in /src/Graph/GraphHelper.cs:line 143
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 238

```

Any ideas?</div>

**Comment by lavinir** [comment]
Are you using a 'Personal' Microsoft account ?
Do you consistently get the error for any upload ? Could you share your configuration ?

**Comment by Operations8** [comment]
> Are you using a 'Personal' Microsoft account ? Do you consistently get the error for any upload ? Could you share your configuration ?

Yes personal. And yes everything i try to upload at the bottom of screen it says "error click here to reload"

```
local_backup_num_to_keep: 0
onedrive_backup_num_to_keep: 5
backup_interval_days: 1
backup_name: hass_onedrive_backup
monitor_all_local_backups: false
ignore_upgrade_backups: false
hass_api_timeout_minutes: 30
notify_on_error: true
exclude_media_folder: true
exclude_ssl_folder: false
exclude_share_folder: false
exclude_local_addons_folder: false
sync_paths: []
file_sync_remove_deleted: false
excluded_addons: []
log_level: info
```

**Comment by lavinir** [comment]
Is it always the same backup failing ? Do you have any other backups you can try to force upload ? (maybe a small partial one)
Please also change the log level to verbose while we're troubleshooting.

**Comment by Operations8** [comment]
> Is it always the same backup failing ? Do you have any other backups you can try to force upload ? (maybe a small partial one) Please also change the log level to verbose while we're troubleshooting.

No all the backups fail.

```
 [1] Verbose: Cleaning up temporary artifcats
 [1] Verbose: Detected Addons: a0d7b954_ssh,core_samba,core_silabs_multiprotocol,a0d7b954_adguard,15d21743_samba_backup,cebe7a76_hassio_google_drive_backup,a0d7b954_nut,a0d7b954_grafana,a0d7b954_vscode,03cabcc9_ring_mqtt,core_mosquitto,ccab4aaf_frigate,de91e161_hassio_onedrive_backup
 [1] Verbose: Ingress URL: /api/hassio_ingress/K-Oe2nm9Tk-NDO5cZMXEKN-Lucf8-fL6uejGnjoShfo/
08/21/2023 09:26:20 [8] Info: Backup interval configured to every 24 hours
08/21/2023 09:26:20 [8] Info: File Sync Disabled
08/21/2023 09:26:23 [6] Verbose: Checking backups
08/21/2023 09:26:23 [6] Verbose: Backup Interval Completed.
08/21/2023 09:26:23 [10] Verbose: Ignoring 'External' backup: Samba Backup 2023-08-21 04:25
08/21/2023 09:26:23 [10] Verbose: Ignoring 'External' backup: bck
08/21/2023 09:26:23 [10] Verbose: Ignoring 'External' backup: Samba Backup 2023-08-15 03:00
08/21/2023 09:26:23 [10] Verbose: Ignoring 'External' backup: hass_backup_2023-08-19-00-16
08/21/2023 09:26:23 [10] Verbose: Ignoring 'External' backup: addon_a0d7b954_vscode_5.10.0
08/21/2023 09:26:26 [10] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: a0d7b954_ssh. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: core_samba. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: core_silabs_multiprotocol. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: a0d7b954_adguard. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: 15d21743_samba_backup. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: cebe7a76_hassio_google_drive_backup. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: a0d7b954_nut. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: a0d7b954_grafana. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: a0d7b954_vscode. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: 03cabcc9_ring_mqtt. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: core_mosquitto. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: ccab4aaf_frigate. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [10] Verbose: Looking for Addon name matching slug: de91e161_hassio_onedrive_backup. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:27 [8] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [6] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [10] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [8] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [11] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [8] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [10] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [11] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [8] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [6] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [10] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [10] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [10] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:27 [11] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:28 [11] Verbose: Source IP: 172.30.32.2
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: a0d7b954_ssh. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: core_samba. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: core_silabs_multiprotocol. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: a0d7b954_adguard. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: 15d21743_samba_backup. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: cebe7a76_hassio_google_drive_backup. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: a0d7b954_nut. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: a0d7b954_grafana. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: a0d7b954_vscode. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: 03cabcc9_ring_mqtt. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: core_mosquitto. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: ccab4aaf_frigate. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:28 [8] Verbose: Looking for Addon name matching slug: de91e161_hassio_onedrive_backup. Checking agaisnt 13 Addons in cache 
08/21/2023 09:26:30 [6] Info: Uploading hass_onedrive_backup_2023-08-19-17-41 (08/19/2023 17:41:23)
08/21/2023 09:26:30 [6] Info: Fetching Local Backup (Slug:111d9886)
08/21/2023 09:26:32 [14] Info: Backup (111d9886) fetched successfully
08/21/2023 09:26:33 [14] Verbose: Backup Description: {"Slug":"111d9886","BackupDate":"2023-08-19T17:41:23.927484+02:00","InstanceName":null,"IsProtected":false,"Size":462.46,"BackupType":"partial","Addons":[],"Folders":["addons/local","share","ssl"]}
08/21/2023 09:26:34 [15] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: invalidRequest
Message: Invalid request
Inner error:
	AdditionalData:
	date: 2023-08-21T07:26:34
	request-id: 9e2b3cb7-e578-47d2-a175-8169d56c22ec
	client-request-id: 9e2b3cb7-e578-47d2-a175-8169d56c22ec
ClientRequestId: 9e2b3cb7-e578-47d2-a175-8169d56c22ec
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at hassio_onedrive_backup.Graph.GraphHelper.UploadFileAsync(String filePath, DateTime date, String instanceName, TransferSpeedHelper transferSpeedHelper, String destinationFileName, Action`2 progressCallback, Boolean flatten, String description) in /src/Graph/GraphHelper.cs:line 143
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 238
08/21/2023 09:26:39 [11] Verbose: Source IP: 172.*.*.2

```

**Comment by Ryckie** [comment]
Hello,

I just freshly installed this addon, and linked it with my second one drive account (1TB free storage) and am also having the issue of no backups uploading. I'm really hyped to use this addon because the Google Drive one for me keeps failing because I have only 15GB storage and I don't want to waste money on increasing it since I have 5TB available on my OneDrive accounts. Also I would like to say thank you for creating this addon for the community!

My configuration:
```local_backup_num_to_keep: 3
onedrive_backup_num_to_keep: 15
backup_interval_days: 1
backup_name: HassioLT_backup
monitor_all_local_backups: true
ignore_upgrade_backups: false
hass_api_timeout_minutes: 30
notify_on_error: true
exclude_media_folder: true
exclude_ssl_folder: false
exclude_share_folder: false
exclude_local_addons_folder: false
sync_paths:
  - /media
file_sync_remove_deleted: false
excluded_addons: []
log_level: verbose
backup_allowed_hours: 6-10
backup_instance_name: Home Assistant LT
backup_passwd: >-
  CensoredMyPassword
```


Console log when restarting the addon:
``` [1] Verbose: Cleaning up temporary artifcats
 [1] Verbose: Detected Addons: core_letsencrypt,a0d7b954_vscode,a0d7b954_ssh,core_samba,core_configurator,cebe7a76_hassio_google_drive_backup,a0d7b954_adguard,a0d7b954_nginxproxymanager,core_mariadb,a0d7b954_wireguard,core_mosquitto,core_zwave_js,6e66619d_room_assistant,a0d7b954_ftp,a0d7b954_spotify,a0d7b954_plex,a0d7b954_bitwarden,a0d7b954_uptime-kuma,core_piper,de91e161_hassio_onedrive_backup
 [1] Verbose: Ingress URL: /api/hassio_ingress/aDkzqeGMhZCby0tBqx0PNelthniYWJslCFpveJDUo3g/
08/25/2023 16:38:27 [8] Info: Backup interval configured to every 24 hours
08/25/2023 16:38:27 [8] Info: Backups / Syncs will only run during these hours: 6,7,8,9,10
08/25/2023 16:38:27 [8] Info: File Sync Enabled
08/25/2023 16:38:31 [6] Verbose: Checking backups
08/25/2023 16:38:31 [6] Verbose: Backup Interval Completed.
08/25/2023 16:38:31 [13] Verbose: Retrieving existing local backups...
08/25/2023 16:38:31 [13] Verbose: Retrieving existing online backups...
08/25/2023 16:38:31 [13] Verbose: Last local backup Date: 08/25/2023 12:02:34
08/25/2023 16:38:31 [13] Verbose: Last online backup Date: None
08/25/2023 16:38:31 [13] Info: Found 3 backups to upload.
08/25/2023 16:38:31 [10] Info: Uploading Full Backup 2023-08-25 12:02:34 (08/25/2023 12:02:34)
08/25/2023 16:38:31 [10] Info: Fetching Local Backup (Slug:57f305f2)
08/25/2023 16:38:35 [6] Info: Backup (57f305f2) fetched successfully
08/25/2023 16:38:44 [6] Verbose: Backup Description: {"Slug":"57f305f2","BackupDate":"2023-08-25T12:02:34.613549+03:00","InstanceName":"Home Assistant LT","IsProtected":true,"Size":342.17,"BackupType":"full","Addons":[],"Folders":["share","addons/local","ssl","media"]}
08/25/2023 16:38:44 [6] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: BadRequest
Message: Multiple action overloads were found with the same binding parameter for 'microsoft.graph.createUploadSession'.
Inner error:
	AdditionalData:
	date: 2023-08-25T13:38:44
	request-id: ac2c11a8-1e74-4006-8552-f3b57bb5486b
	client-request-id: ac2c11a8-1e74-4006-8552-f3b57bb5486b
ClientRequestId: ac2c11a8-1e74-4006-8552-f3b57bb5486b
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at hassio_onedrive_backup.Graph.GraphHelper.UploadFileAsync(String filePath, DateTime date, String instanceName, TransferSpeedHelper transferSpeedHelper, String destinationFileName, Action`2 progressCallback, Boolean flatten, String description) in /src/Graph/GraphHelper.cs:line 143
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 238
08/25/2023 16:38:48 [6] Info: Uploading Full Backup 2023-08-24 12:00:00 (08/24/2023 12:00:00)
08/25/2023 16:38:48 [6] Info: Fetching Local Backup (Slug:b845977f)
08/25/2023 16:38:51 [6] Info: Backup (b845977f) fetched successfully
08/25/2023 16:38:59 [6] Verbose: Backup Description: {"Slug":"b845977f","BackupDate":"2023-08-24T12:00:00.192373+03:00","InstanceName":"Home Assistant LT","IsProtected":true,"Size":340.52,"BackupType":"full","Addons":[],"Folders":["share","addons/local","ssl","media"]}
08/25/2023 16:38:59 [6] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: BadRequest
Message: Multiple action overloads were found with the same binding parameter for 'microsoft.graph.createUploadSession'.
Inner error:
	AdditionalData:
	date: 2023-08-25T13:38:59
	request-id: dbb57522-443c-47c5-8892-db9b2a6b78ea
	client-request-id: dbb57522-443c-47c5-8892-db9b2a6b78ea
ClientRequestId: dbb57522-443c-47c5-8892-db9b2a6b78ea
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at hassio_onedrive_backup.Graph.GraphHelper.UploadFileAsync(String filePath, DateTime date, String instanceName, TransferSpeedHelper transferSpeedHelper, String destinationFileName, Action`2 progressCallback, Boolean flatten, String description) in /src/Graph/GraphHelper.cs:line 143
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 238
08/25/2023 16:39:01 [6] Info: Uploading Full Backup 2023-08-23 12:00:00 (08/23/2023 12:00:00)
08/25/2023 16:39:01 [6] Info: Fetching Local Backup (Slug:d6157107)
08/25/2023 16:39:04 [11] Info: Backup (d6157107) fetched successfully
08/25/2023 16:39:12 [11] Verbose: Backup Description: {"Slug":"d6157107","BackupDate":"2023-08-23T12:00:00.529962+03:00","InstanceName":"Home Assistant LT","IsProtected":false,"Size":340.2,"BackupType":"full","Addons":[],"Folders":["share","addons/local","ssl","media"]}
08/25/2023 16:39:12 [11] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: BadRequest
Message: Multiple action overloads were found with the same binding parameter for 'microsoft.graph.createUploadSession'.
Inner error:
	AdditionalData:
	date: 2023-08-25T13:39:12
	request-id: 734028cb-6f29-47a4-8008-871f8369784d
	client-request-id: 734028cb-6f29-47a4-8008-871f8369784d
ClientRequestId: 734028cb-6f29-47a4-8008-871f8369784d
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at hassio_onedrive_backup.Graph.GraphHelper.UploadFileAsync(String filePath, DateTime date, String instanceName, TransferSpeedHelper transferSpeedHelper, String destinationFileName, Action`2 progressCallback, Boolean flatten, String description) in /src/Graph/GraphHelper.cs:line 143
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 238```

**Comment by Ryckie** [comment]
I have now tried with a different configuration and this time it did not upload my backups that we're existing. I triggered one manually in the UI from this addon and that one uploaded with success. When I change to not ignore the local backups it gives me the errors again and doesn't upload anymore:
```local_backup_num_to_keep: 4
onedrive_backup_num_to_keep: 20
backup_interval_days: 1
backup_name: hass_backup
monitor_all_local_backups: false
ignore_upgrade_backups: false
hass_api_timeout_minutes: 30
notify_on_error: true
exclude_media_folder: true
exclude_ssl_folder: false
exclude_share_folder: false
exclude_local_addons_folder: false
sync_paths:
  - /media
file_sync_remove_deleted: false
excluded_addons: []
log_level: info
backup_allowed_hours: 6-9
backup_instance_name: HassioLT
backup_passwd: >-
  HiddenMyPsswd
```

**Comment by lavinir** [comment]
There are some restrictions in OneDrive with things like some characters they do not allow in the name. Errors are unfortunately not very clear sometimes. Glad it's working now - If there are any other issues let me know.

**Comment by Operations8** [comment]
Still not working for me. The startpost is mine.

**Comment by lavinir** [comment]
I tried reproducing your settings but I'm not getting any errors. Could you try using the File Sync to upload a file(s) and see if that's successful ?

**Comment by Operations8** [comment]
Not sure what file sync is but i will check. Upload to onedrive you mean? Just a reminder the Google drive backup app does work, as does Samba backup to my NAS.

**Comment by Ryckie** [comment]
> Not sure what file sync is but i will check. Upload to onedrive you mean? Just a reminder the Google drive backup app does work, as does Samba backup to my NAS.

Hey, could you try deleting all local backups and triggering a new one in the add-on and see if it works then. That's what did it for me with similar errors.

The sync is to sync specific folders seperate so you can have it excluded from backups and making them take less space.
Example:
exclude_media_folder: true
sync_paths:
  - /media

**Comment by Operations8** [comment]
I removed all backups and opened OneDrive, the error was only a few seconds there but it said something about HTML filtering.

**Comment by Operations8** [comment]
I created a page rule now in Cloudflare. Would something about HTML filtering in the FAQ

**Comment by Operations8** [comment]
Side question why is only MS personal account possible? I pay for M365 that gives me 1TB of Onedrive space i cannot use right now... plans to add Microsoft 365 not personal accounts option?

**Comment by Operations8** [comment]
With page rule it still doesnt work.

**Comment by lavinir** [comment]
1. Rule for Cloudflare is relevant for the UI to function properly but is not related to backup failures 
2. The reason only a Personal Onedrive account is supported for now is because this addon is using a special permission scope that limits its permissions to your OneDrive data to a dedicated App folder only accessible by the Addon. This was mandatory for me as I do not want access to your entire OneDrive data. This permissions is still not available for OneDrive for Business (more info [here ](https://learn.microsoft.com/en-us/onedrive/developer/rest-api/concepts/special-folders-appfolder?view=odsp-graph-online#getting-authorization-from-the-user)if you're interested)

More information about using File Sync capabilities is [here](https://github.com/lavinir/hassio-onedrive-backup#file-sync-paths-optional). Please try and sync something and let me know if that has also failed.

**Comment by Operations8** [comment]
I already used file sync (i added /media in the options). I removed all local backups first.

New log:

```

08/27/2023 09:41:52 [11] Info: Backup interval configured to every 24 hours
08/27/2023 09:41:52 [11] Info: File Sync Enabled
08/27/2023 09:46:00 [3] Info: Creating new backup
08/27/2023 09:46:00 [3] Info: Starting partial local backup
08/27/2023 09:46:57 [31] Info: Backup complete
08/27/2023 09:47:05 [32] Info: Uploading hass_onedrive_backup_2023-08-27-09-46 (08/27/2023 09:46:00)
08/27/2023 09:47:05 [32] Info: Fetching Local Backup (Slug:37fd0731)
08/27/2023 09:47:06 [32] Info: Backup (37fd0731) fetched successfully
08/27/2023 09:47:08 [31] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: invalidRequest
Message: Invalid request
Inner error:
	AdditionalData:
	date: 2023-08-27T07:47:08
	request-id: 7b388fda-a7b7-43c3-bcfb-9f86155f7fcf
	client-request-id: 7b388fda-a7b7-43c3-bcfb-9f86155f7fcf
ClientRequestId: 7b388fda-a7b7-43c3-bcfb-9f86155f7fcf
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at hassio_onedrive_backup.Graph.GraphHelper.UploadFileAsync(String filePath, DateTime date, String instanceName, TransferSpeedHelper transferSpeedHelper, String destinationFileName, Action`2 progressCallback, Boolean flatten, String description) in /src/Graph/GraphHelper.cs:line 143
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 238

```

**Comment by lavinir** [comment]
This is still the backup upload failure. For the File Sync test you need to actually have some file(s) in the path you specified to sync. 
I want to see if any file can be uploaded (to understand if this is a global issue or something backup specific)

**Comment by Operations8** [comment]
> This is still the backup upload failure. For the File Sync test you need to actually have some file(s) in the path you specified to sync. I want to see if any file can be uploaded (to understand if this is a global issue or something backup specific)

There are many files in /media folder.

**Comment by lavinir** [comment]
Then we're missing that part of the log. Look for when it starts the File Sync iteration and tries to upload files.

**Comment by Operations8** [comment]
> Then we're missing that part of the log. Look for when it starts the File Sync iteration and tries to upload files.

The only time the word "media" appears in the log is here: (when you click on details of the created backup in the one drive backup GUI you says the folders stated)

```
08/29/2023 19:01:10 [86] Verbose: Backup Description: {"Slug":"379a0ee0","BackupDate":"2023-08-29T18:55:39.062211+02:00","InstanceName":null,"IsProtected":false,"Size":750.22,"BackupType":"full","Addons":[],"Folders":["share","addons/local","ssl","media"]}
08/29/2023 19:01:11 [84] Error: Error uploading backup: Status Code: BadRequest
Microsoft.Graph.ServiceException: Code: invalidRequest
Message: Invalid request
Inner error:
	AdditionalData:
	date: 2023-08-29T17:01:11
	request-id: 44cf06b8-8e81-422f-b3ff-853212a9858c
	client-request-id: 44cf06b8-8e81-422f-b3ff-853212a9858c
ClientRequestId: 44cf06b8-8e81-422f-b3ff-853212a9858c
   at 

```

**Comment by mariobrosch** [comment]
same badrequest error here. A few partially created backups work just fine. The full backup created does not upload properly. As mentioned because of the invalid character (https://support.microsoft.com/en-gb/office/restrictions-and-limitations-in-onedrive-and-sharepoint-64883a5d-228e-48f5-b3d2-eb39e07630fa#invalidcharacters)

The invalid name was created by a backup tool which syncs to google drive, so removed the backup and recreated via the onedrive addon and it works great. I suggest changing the addon so it removes invalid characters is a safe way to go.

**Comment by lavinir** [comment]
fixed in 2.2
